### DISCUSSION TITLE
FDTD for bent surfaces

### DISCUSSION THREAD

**Discussion by nico-arnold** [thliebig/openEMS-Project#106] [Q&A Usage] [answered]
Hello everybody,

for a project, I need to make some simulations on bent pieces of flexible PCB. I'm not completely sure if that is possible to do via the FDTD method and how to mesh that. The geometry to be simulated is a U-shaped bend that contains test traces configured as a coplanar waveguide, and I need to calculate the field distribution and the wave impedance of this one:
![image](https://github.com/thliebig/openEMS-Project/assets/82146516/630d88d9-9421-4fa6-b307-4b4609007b5c)

Prior experiments with the same geometry in a flat state seemed to work pretty well:
![image](https://github.com/thliebig/openEMS-Project/assets/82146516/b8d1fc1b-0819-41c0-9377-604de135ab06)
For this one I took the transmission line example from the example projects and added some more lines in the area of the bends so that the line density is the same there as in the edge regions of the straight trace from the example.

However, when working on the bent geometry, the calculation takes very long due to the fine grid, and in the end the signal never "jumps" from the ports onto the PCB trace. It just is generated on the port, travels along this one and decays there instead of being conducted onto the waveguide.
I would be very happy for any recommendations or tips here! (:

best wishes,
Nico

PS: Files can be found at (https://cloudstore.zih.tu-dresden.de/index.php/s/Hwo2J7NNSe7HQYy)

**Comment by thliebig** [comment, marked as answer]
Well I think the FDTD method in general will struggle with this setup and openEMS in particular as it lacks some functionality to make sure that metal traces get mapped to the grid well enough for such sharp bents and thin metals...

**Reply by XDAChen** [reply]
#thliebig, I am also having similar issues: I wanted to simulate a bent electrical component, also a thin metal, but struggle to get a reasonable simulation result.

It seems like there are two issues here: 1) thin metal and 2) sharp bend. I may understand reason 1) since thin metal requires super high grained mesh to accurately simulate. But why is sharp bend an issue with FDTD/openEMS? 

If one can manually (or contribute a function) segregate the yee cells at the edge of the metal by two-third in a 3D manner, do you think it would make the issue go away? In other words, do you think there are other reasons that would make the simulation of bent-thin metal unfavorable using openEMS other than the two mentioned above?

**Reply by thliebig** [reply]
Well you always have to have a close look how your metal is mapped to the grid. Use the debugging option "--debug-PEC" for that.
But in short a close surface of Yee cell edges have to be assigned to metal for a signal line to work. And on a bend or conformal structure that is very difficult and openEMS currently has no good surface mapping for metals for that. This is not an easy algorithm to implement (mapping a arbitrary 3D metal surface to edge grid).

**Reply by Sarajevo67** [reply]
> This is not an easy algorithm to implement (mapping a arbitrary 3D metal surface to edge grid).

Is it possible to solve this problem (solid 3D surfaces, not thin metal), using an approximation, similar to the 1/3 2/3 rule, or is it limited only to 2D surfaces?

By the way, you posted a link somewhere to a document that addresses the rule of thirds, but I can't seem to find it?

**Comment by nico-arnold** [comment]
Ok, that's what I already feared. Still, thank you very much for your help! I will keep OpenEMS then for the flat structures as it's a great piece of software, and focus on other tools then for the bent structures. (:
